File: Algorithms/Implementation/Larry's Array.php
<?php

/*
 * Complete the 'larrysArray' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts INTEGER_ARRAY A as parameter.
 */

function larrysArray($A) {
    # Write your code here
    $n = count($A);
    $inversions = 0;

    for ($i = 0; $i < $n - 1; $i++) {
        for ($j = $i + 1; $j < $n; $j++) {
            if ($A[$i] > $A[$j]) {
                $inversions++;
            }
        }
    }

    return ($inversions % 2 == 0) ? "YES" : "NO";
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$t = intval(trim(fgets(STDIN)));

for ($t_itr = 0; $t_itr < $t; $t_itr++) {
    $n = intval(trim(fgets(STDIN)));

    $A_temp = rtrim(fgets(STDIN));

    $A = array_map('intval', preg_split('/ /', $A_temp, -1, PREG_SPLIT_NO_EMPTY));

    $result = larrysArray($A);

    fwrite($fptr, $result . "\n");
}

fclose($fptr);

File: Algorithms/Implementation/Larry's Array.py
import os

#
# Complete the 'larrysArray' function below.
#
# The function is expected to return a STRING.
# The function accepts INTEGER_ARRAY A as parameter.
#

def larrysArray(A):
    # Write your code here
    def count_inversions(arr):
        inv_count = 0
        for i, val_i in enumerate(arr):
            for j, val_j in enumerate(arr[i + 1 :], start=i + 1):
                if val_i > val_j:
                    inv_count += 1
        return inv_count

    inversions = count_inversions(A)
    if inversions % 2 == 0:
        return "YES"
    else:
        return "NO"


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input().strip())

    for t_itr in range(t):
        n = int(input().strip())

        A = list(map(int, input().rstrip().split()))

        result = larrysArray(A)

        fptr.write(result + '\n')

    fptr.close()
